libs/webb-ui-components/src/components/Pagination/Pagination.tsx

```diff
@@ -38,11 +38,20 @@ export const Pagination = React.forwardRef<HTMLDivElement, PaginationProps>(
       [boundaryCount, currentPage, siblingCount, totalPages],
     );
 
-    const showingItemsCount = useMemo(
-      () =>
-        itemsPerPage && totalItems ? Math.min(itemsPerPage, totalItems) : '-',
-      [itemsPerPage, totalItems],
-    );
+    const showingItemsCount = useMemo(() => {
+      const isLastPage = currentPage === totalPages;
+
+      // If not the last page, return the itemsPerPage
+      if (!isLastPage) {
+        return itemsPerPage?.toLocaleString() ?? '-';
+      }
+
+      // Otherwise, calculate the remaining items on the last page
+      const remainingItems = totalItems ? totalItems % (itemsPerPage ?? 1) : 0;
+      return remainingItems > 0
+        ? remainingItems.toLocaleString()
+        : (itemsPerPage?.toLocaleString() ?? '-');
```

Comment:
Should use `EMPTY_VALUE_PLACEHOLDER` instead of `-`?

Reply:
`EMPTY_VALUE_PLACEHOLDER` is part of the `tangle-dapp`, but this component belongs to `webb-ui-components`, making the constant unable to import here.